File: lab/javascript/matematica_e_operadores/contagem_regressiva.js
function contagem_regressiva_for(numero){
    let lista = [];
    for(let i = numero; i >= 0; i--){
        lista.push(i);
    }
    return join(lista, ' ');
}


function contagem_regressiva_while(numero){
    let lista = [];
    while(numero >= 0){
        lista.push(numero);
        numero--;
    }
    return join(lista, ' ');
}

function contagem_regressiva_do_while(numero){
    let lista = [];
    do{
        lista.push(numero);
        numero--;
    }while(numero >= 0);
    return join(lista, ' ');
}

function join(lista, separador){
    let resultado = '';
    for(let i = 0; i < lista.length; i++){
        resultado += lista[i];
        if(i < lista.length - 1){
            resultado += separador;
        }
    }
    return resultado;
}

console.log(contagem_regressiva_for(5));
console.log(contagem_regressiva_while(5));
console.log(contagem_regressiva_do_while(5));

File: lab/c/matematica_e_operadores/contagem_regressiva.c
#include <stdio.h>

void contagem_regressiva_for(int numero);
void contagem_regressiva_while(int numero);
void contagem_regressiva_dowhile(int numero);

int main() {
    contagem_regressiva_for(5);
    contagem_regressiva_while(5);
    contagem_regressiva_dowhile(5);
    return 0;
}

void contagem_regressiva_for(int numero) {
    for (int i = numero; i >= 0; i--) {
        printf("%d ", i);
    }
    printf("\n");
}

void contagem_regressiva_while(int numero) {
    int i = numero;
    while (i >= 0) {
        printf("%d ", i);
        i--;
    }
    printf("\n");
}

void contagem_regressiva_dowhile(int numero) {
    int i = numero;
    do {
        printf("%d ", i);
        i--;
    } while (i >= 0);
    printf("\n");
}
